src/relations/postgresql.py

```diff
@@ -88,6 +74,62 @@ def _on_database_relation_broken(self, event: DatabaseEvent) -> None:
         self._update_db_connections(event.relation.name, None)
         self.charm._update(event)
 
+    def update_db_relation_data_in_state(self) -> bool:
+        """Update database data from relation into peer relation databag.
+
+        Returns:
+            True if the charm should update its pebble layer, False otherwise.
+        """
+        if not self.charm.unit.is_leader():
+            return False
```

Comment:
suggestion: Raise exceptions instead of returning false in this section. No other way to propagate that an event was skipped.

Also, I don't see a defer. What happens if the `is_ready()` returns false? I think it would skip the hook permanently.

Reply:
The `False` is returned to signify to the charm that it should not re-plan the pebble layer yet. I've added the defer, but it will not skip the event permanently given that it retries periodically on the `update_status` hook.